netbox/dcim/choices.py

```diff
@@ -1399,3 +1399,74 @@ class PowerFeedPhaseChoices(ChoiceSet):
         (PHASE_SINGLE, 'Single phase'),
         (PHASE_3PHASE, 'Three-phase'),
     )
+
+
+#
+# VDC
+#
+class VirtualDeviceContextStatusChoices(ChoiceSet):
+    key = 'VirtualDeviceContext.status'
+
+    STATUS_PLANNED = 'planned'
+    STATUS_ACTIVE = 'active'
+    STATUS_OFFLINE = 'offline'
+
+    CHOICES = [
+        (STATUS_PLANNED, 'Planned', 'cyan'),
+        (STATUS_ACTIVE, 'Active', 'green'),
+        (STATUS_OFFLINE, 'Offline', 'red'),
+    ]
+
+
+class VirtualDeviceContextTypeChoices(ChoiceSet):
```

Comment:
If we do end up needing to track the VDC type, it would probably make sense for these to be user-defined, as we can't reasonably anticipate every vendor implementation. (However as I commented elsewhere we can probably just infer the VDC type from the assigned platform.)

Reply:
Instead of using this..  Could we instead have a field for "Allow interface sharing within virtual device contexts" on the device type?  Default it to none, which would be either Yes or No.